phys2denoise.py

```diff
@@ -0,0 +1,234 @@
+
+
+def select_input_args(metric, metric_args):
+    """
+    Retrieve required args for metric from a dictionary of possible arguments.
+
+    This function checks what parameters are accepted by a metric.
+    Then, for each parameter, check if the user provided it or not.
+    If they did not, but the parameter is required, throw an error -
+    unless it's "physio", reserved name for the timeseries input to a metric.
+    Otherwise, use the default.
+
+    Parameters
+    ----------
+    metric : function
+        Metric function to retrieve arguments for
+    metric_args : dict
+        Dictionary containing all arguments for all functions requested by the
+        user
+
+    Returns
+    -------
+    args : dict
+        Arguments to provide as input to metric
+
+    Raises
+    ------
+    ValueError
+        If a required argument is missing
+
+    """
+    args = {}
+
+    # Check the parameters required by the metric and given by the user (see docstring)
+    for param in signature(metric).parameters.values():
+        if param.name not in metric_args:
+            if param.default == _empty and param.name != 'physio':
+                raise ValueError(f'Missing parameter {param} required '
+                                 f'to run {metric}')
+            else:
+                args[param.name] = param.default
+        else:
+            args[param.name] = metric_args[param.name]
+
+    return args
+
+
+def print_metric_call(metric, args):
+    """
+    Log a message to describe how a metric is being called.
+
+    Parameters
+    ----------
+    metric : function
+        Metric function that is being called
+    args : dict
+        Dictionary containing all arguments that are used to parametrise metric
+
+    Notes
+    -----
+    Outcome
+        An info-level message for the logger.
+    """
+    msg = f'The {metric} regressor will be computed using the following parameters:'
+
+    for arg in args:
+        msg = f'{msg}\n    {arg} = {args[arg]}'
+
+    msg = f'{msg}\n'
+
+    LGR.info(msg)
+
+
+@due.dcite(
+     Doi(''),
+     path='phys2denoise',
+     description='Creation of regressors for physiological denoising',
+     version=__version__,
+     cite_module=True)
+def phys2denoise(filename, metric_args, outdir='.',
+                 metrics=[crf, rpv, rv, rvt, rrf, 'retroicor_card', 'retroicor_resp'],
+                 debug=False, quiet=False):
+    """
+    Run main workflow of phys2denoise.
+
+    Runs the parser, does some checks on input, then computes the required metrics.
+
+    Notes
+    -----
+    The code was greatly copied from phys2bids (copyright the physiopy community)
+
+    """
+    # Check options to make them internally coherent pt. I
+    # #!# This can probably be done while parsing?
+    outdir = os.path.abspath(outdir)
+    os.makedirs(outdir)
+    os.makedirs(os.path.join(outdir, 'code'))
+    log_path = os.path.join(outdir, 'code', 'logs')
+    os.makedirs(log_path)
```

Comment:
I don't think there's any need to call `makedirs` repeatedly. It creates parent directories automatically, so you can just have the last one, right?

Reply:
I would say so, yes.